File: C2J/sync/Multi-Level-1/multilevel.js
jcond{
    isFog: sys.type == "fog";
    isDevice: sys.type == "device";
}

var deviceId = 1000;
var PROCESS_COUNT = 1;

// jsync function to assign id's to devices
jsync {isFog} function getId() {
    var id = ++deviceId;
    return id + "";
}

jsync {isDevice} function getPayload() {
    return PROCESS_COUNT;
}

File: C2J/sync/Multi-Level-1/multilevel.c
#include <string.h>
#include <stdlib.h>
#include <stdio.h>
#include <unistd.h>
#include <sys/time.h>

char* getId();
int getPayload();


char* nodeID;
int PROCESS_COUNT;  //number of data item points to process;



int main(int argc, char** argv){

    printf("C is running...\n");

    nodeID = getId();
    PROCESS_COUNT = getPayload();

    printf("PID is %d My ID is %s\n", getpid(), nodeID);

}
